### PR TITLE
[Docs ] Tune docs overhaul (first part)

### PR DESCRIPTION
**Pull request by maxpumperla**
Preview: [Tune docs](https://ray--22112.org.readthedocs.build/en/22112/tune/index.html)

Continuing docs overhaul, tune now has:

- [x] better landing page
- [x] a getting started guide
- [x] user guide was cut down, partially merged with FAQ, and partially integrated with tutorials
- [x] the new user guide contains guides to tune features and practical integrations
- [x] we rewrote some of the feature guides for clarity 
- [x] we got rid of sphinx-gallery for this sub-project (only data and core left), as it looks bad and is unnecessarily complicated anyway (plus, makes the build slower)
- [x] sphinx-gallery examples are now moved to markdown notebook, as started in #22030.
- [x] Examples are tested in the new framework, of course.

There's still a lot one can do, but this is already getting too large. Will follow up with more fine-tuning next week.

### CODE REVIEW

File: doc/source/tune/faq.rst [hunk 1]
```diff
@@ -516,3 +516,330 @@ should maybe process a larger chunk of data. In function trainables, you can rep
...
+        # ...,
+        local_dir="/path/to/shared/storage",
+        sync_config=tune.SyncConfig(
+            # Do not sync because we are on shared storage
+            syncer=None
+        )
+    )
+
+
+Lastly, if you still want to use ssh for trial synchronization, but are not running
```
Comment (Yard1) [inline doc/source/tune/faq.rst:763]:
```suggestion
Lastly, if you still want to use SSH for trial synchronization, but are not running
```

File: doc/source/tune/faq.rst [hunk 2]
```diff
@@ -516,3 +516,330 @@ should maybe process a larger chunk of data. In function trainables, you can rep
...
+Local mode with multiple configuration evaluations will interleave computation,
+so it is most naturally used when running a single configuration evaluation.
+
+Note that ``local_mode`` has some known issues, so please read :ref:`these tips <local-mode-tips>` for more info.
+
+
+
+.. _tune-default-search-space:
+
+How Do I Configure Search Spaces?
```
Comment (Yard1) [inline doc/source/tune/faq.rst:805]:
```suggestion
How do I configure search spaces?
```

File: doc/source/tune/faq.rst [hunk 3]
```diff
@@ -516,3 +516,330 @@ should maybe process a larger chunk of data. In function trainables, you can rep
...
+            "nn_layers": [
+                tune.grid_search([16, 64, 256]),
+                tune.grid_search([16, 64, 256]),
+            ],
+        },
+        num_samples=10
+    )
+
+Note that search spaces may not be interoperable across different search algorithms.
+For example, for many search algorithms, you will not be able to use a ``grid_search`` parameter.
```
Comment (Yard1) [inline doc/source/tune/faq.rst:844]:
```suggestion
For example, for many search algorithms, you will not be able to use a ``grid_search`` or ``sample_from`` parameters.
```

File: doc/source/tune/index.rst [hunk 4]
```diff
@@ -33,67 +98,57 @@ This example runs a parallel grid search to optimize an example objective functi
...
+.. dropdown:: Coming From Another Hyperparameter Optimization Tool?
 
-You might be already using an existing hyperparameter tuning tool such as HyperOpt or Bayesian Optimization.
+    You might be already using an existing hyperparameter tuning tool such as HyperOpt or Bayesian Optimization.
 
-In this situation, Tune actually allows you to power up your existing workflow. Tune's :ref:`Search Algorithms <tune-search-alg>` integrate with a variety of popular hyperparameter tuning libraries (such as Nevergrad or HyperOpt) and allow you to seamlessly scale up your optimization process -- without sacrificing performance.
+    In this situation, Tune actually allows you to power up your existing workflow.
+    Tune's :ref:`Search Algorithms <tune-search-alg>` integrate with a variety of popular hyperparameter tuning
+    libraries (such as Nevergrad or HyperOpt) and allow you to seamlessly scale up your optimization
+    process -- without sacrificing performance.
```
Comment (Yard1) [inline doc/source/tune/index.rst:144]:
```suggestion
    process - without sacrificing performance.
```

File: doc/source/tune/index.rst [hunk 5]
```diff
@@ -33,67 +98,57 @@ This example runs a parallel grid search to optimize an example objective functi
...
-A key problem with machine learning frameworks is the need to restructure all of your code to fit the framework.
+    A key problem with machine learning frameworks is the need to restructure all of your code to fit the framework.
 
-With Tune, you can optimize your model just by :ref:`adding a few code snippets <tune-tutorial>`.
+    With Tune, you can optimize your model just by :ref:`adding a few code snippets <tune-tutorial>`.
 
-Further, Tune actually removes boilerplate from your code training workflow, automatically :ref:`managing checkpoints <tune-checkpoint-syncing>` and :ref:`logging results to tools <tune-logging>` such as MLflow and TensorBoard.
+    Also, Tune actually removes boilerplate from your code training workflow,
+    automatically :ref:`managing checkpoints <tune-checkpoint-syncing>` and
+    :ref:`logging results to tools <tune-logging>` such as MLflow and TensorBoard.
```
Comment (Yard1) [inline doc/source/tune/index.rst:124]: Should we mention something about how customisable Tune is here?

**Review by Yard1 — COMMENTED** [review]
Gave it a pass. I really love the landing page, and the new guides are awesome.

File: doc/source/tune/tutorials/tune-stopping.rst [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+
+If you send a SIGINT signal to the process running ``tune.run()`` (which is
+usually what happens when you press Ctrl+C in the console), Ray Tune shuts
+down training gracefully and saves a final experiment-level checkpoint.
+
+How to resume a Tune run?
+-------------------------
+
+If you've stopped a run and and want to resume from where you left off,
+you can then call ``tune.run()`` with ``resume=True`` like this:
```
[the remark this answers is not in the document]
Comment (maxpumperla) [reply doc/source/tune/tutorials/tune-stopping.rst:16]: We could do that, but on that note I just want to mention that this option currently doesn't exist in our API docs either:

```
resume (str|bool) – One of “LOCAL”, “REMOTE”, “PROMPT”, “ERRORED_ONLY”, or bool. LOCAL/True restores the checkpoint from the local experiment directory, determined by name and local_dir. REMOTE restores the checkpoint from upload_dir (as passed to sync_config). PROMPT provides CLI feedback. False forces a new experiment. ERRORED_ONLY resets and reruns ERRORED trials upon resume - previous trial artifacts will be left untouched. If resume is set but checkpoint does not exist, ValueError will be thrown.
```

So at least we should add that there, too.

[the remark this answers is not in the document]
Comment (maxpumperla) [reply doc/source/tune/tutorials/tune-stopping.rst:16]: ok, I've added it in both places

File: doc/source/tune/tutorials/tune-output.rst [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+
+To learn more about Trials, see its detailed API documentation: :ref:`trial-docstring`.
+
+.. _tensorboard:
+
+How to log to TensorBoard?
+--------------------------
+
+Tune automatically outputs Tensorboard files during ``tune.run``.
+To visualize learning in tensorboard, install tensorboardX:
```
[the remark this answers is not in the document]
Comment (maxpumperla) [reply doc/source/tune/tutorials/tune-output.rst:62]: done, made it consistent across all docs now.

File: doc/source/tune/tutorials/tune-lifecycle.rst [hunk 8]
```diff
@@ -68,11 +66,12 @@ If the trainable is a callable or a function, it will be executed on the Ray act
 Resource Management in Tune
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Before running a trial, the Tune Driver will check whether there are available resources on the cluster (see :ref:`resource-requirements`). It will compare the available resources with the resources required by the trial.
-
-If there is space on the cluster, then the Tune Driver will start a Ray actor (worker). This actor will be scheduled and executed on some node where the resources are available.
+Before running a trial, the Tune Driver will check whether there are available resources on the cluster (see :ref:`resource-requirements`).
```
Comment (krfricke) [inline doc/source/tune/tutorials/tune-lifecycle.rst:69]:
```suggestion
Before running a trial, the Ray Tune driver will check whether there are available resources on the cluster (see :ref:`resource-requirements`).
```

This is a general thing and not related to this PR (and could be done in a follow up), but generally we want to use "Ray Tune" everywhere we currently use only "Tune"

**Review by krfricke — COMMENTED** [review]
This is great, thank you so much for this refactoring.

As a general question, can we HTTP redirect from the deleted pages to some of the new pages? I want to avoid people getting lost on 404 errors. This might be something we have to change in a different location though.

**Comment by maxpumperla** [comment]
@krfricke thanks! yes, we can define redirect rules directly in readthedocs.org

**Review by krfricke — APPROVED** [review]
Awesome!!

**Pull request merged**